File: Src/Icm.Core/Search and Replace/AutoNumberGenerator.vb
Namespace Icm.Text

    ''' <summary>
    ''' Returns the natural numbers, starting with 1
    ''' </summary>
    ''' <remarks></remarks>
    Public Class AutoNumberGenerator
        Implements IEnumerator(Of String)


        Private counter_ As Integer
        Public Sub New()
            counter_ = 0
        End Sub


        Public ReadOnly Property Current As String Implements IEnumerator(Of String).Current
            Get
                Return CStr(counter_)
            End Get
        End Property

        Public ReadOnly Property Current1 As Object Implements IEnumerator.Current
            Get
                Return Current
            End Get
        End Property

        Public Function MoveNext() As Boolean Implements IEnumerator.MoveNext
            counter_ = counter_ + 1
        End Function

        Public Sub Reset() Implements IEnumerator.Reset
            Throw New NotImplementedException
        End Sub


        Public Sub Dispose() Implements IDisposable.Dispose
        End Sub

    End Class

End Namespace

File: Src/Icm.Core/Search and Replace/AutoNumberGenerator.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace Icm.Text
{

	/// <summary>
	/// Returns the natural numbers, starting with 1
	/// </summary>
	/// <remarks></remarks>
	public class AutoNumberGenerator : IEnumerator<string>
	{



		private int counter_;
		public AutoNumberGenerator()
		{
			counter_ = 0;
		}


		public string Current {
			get { return Convert.ToString(counter_); }
		}

		public object Current1 {
			get { return Current; }
		}
		object IEnumerator.Current {
			get { return Current1; }
		}

		public bool MoveNext()
		{
			counter_ = counter_ + 1;
			return false;
		}

		public void Reset()
		{
			throw new NotImplementedException();
		}


		public void Dispose()
		{
		}

	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
